drivers/flash/flash_stm32_qspi.c

```diff
@@ -639,9 +840,66 @@ static int spi_nor_process_bfp(const struct device *dev,
 
 	LOG_DBG("Page size %u bytes", data->page_size);
 	LOG_DBG("Flash size %u bytes", flash_size);
+
+	addr_mode = jesd216_bfp_addrbytes(bfp);
+	if (addr_mode == JESD216_SFDP_BFP_DW1_ADDRBYTES_VAL_3B4B) {
+		struct jesd216_bfp_dw16 dw16;
+
+		if (jesd216_bfp_decode_dw16(php, bfp, &dw16) == 0) {
+			/*
+			 * According to JESD216, the bit0 of dw16.enter_4ba
+			 * portion of flash description register 16 indicates
+			 * if it is enough to use 0xB7 instruction without
+			 * write enable to switch to 4 bytes addressing mode.
+			 */
+			if (dw16.enter_4ba & 0x1) {
+				rc = qspi_program_addr_4b(dev);
+				if (rc == 0) {
+					data->flag_access_32bit = true;
+					LOG_INF("Flash - address mode: 4B");
+				} else {
+					LOG_ERR("Unable to enter 4B mode: %d\n",
+						rc);
+					return rc;
+				}
+			}
+		}
+	}
+	/*
+	 * Only check if the 1-4-4 (i.e. 4READ) fast read operation is
+	 * supported - other modes - e.g. 1-1-4 (QREAD) or 1-1-2 (DREAD) are
+	 * not.
+	 */
+	struct jesd216_instr res;
+
+	rc = jesd216_bfp_read_support(php, bfp, JESD216_MODE_144, &res);
+	if (IS_ENABLED(STM32_QSPI_QUAD_IO) && rc > 0) {
+		/* Program flash memory to use SIO[0123] */
+		qspi_program_quad_io(dev);
```

Comment:
The function can fail and report that by returning in band error codes, so shouldn't this be checked here? Before doing anything else below?

Reply:
+1